nestkernel/connection_creator_impl.h

```diff
@@ -757,6 +757,8 @@ ConnectionCreator::fixed_outdegree_( Layer< D >& source,
         rng_delay_vec.push_back( delay_[ indx ]->value( grng, source_pos_vector, target_pos_vector, target, tgt ) );
       }
 
+      --number_of_connections;
+
       // We bail out for non-local neurons only now after all possible
       // random numbers haven been drawn. Bailing out any earlier may lead
       // to desynchronized global rngs.
```

Comment:
We should comment a bit better what is going on here. The `if ( not ... )` line should come right after the `--number_...` line.
```suggestion
      // Each VP has now decided to create this connection and drawn any random parameter values
      // required for it. Each VP thus counts the connection as created, but only the VP hosting the
      // target neuron actually creates the connection.
      --number_of_connections;
```